### PR TITLE
Add Evergreen input field component

### PR DESCRIPTION
**Pull request by mparent61**

### CODE REVIEW

File: app/components/evergreen-field/evergreen-field-directive.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+angular.module('evergreenField.directive', [
+  'bulbsCmsApp.settings',
+  'lodash',
+  'saveButton.directive',
+])
+  .directive('evergreenField',
+    // TODO: akos -- this breaks when using new-style array, "routes" is undefined
+    function (routes) {
```
Comment (kand) [inline app/components/evergreen-field/evergreen-field-directive.js]: Set it up like this:
```js
'use strict';

angular.module('evergreenField.directive', [
  'bulbsCmsApp.settings',
  'lodash',
  'saveButton.directive',
])
  .directive('evergreenField', [
    'routes',
    function (routes) {
      return {
        controller: [
          '_', '$scope', 'ContentFactory',
          function (_, $scope, ContentFactory) {
        }],
        restrict: 'E',
        scope: {
          content: '='
        },
        templateUrl: routes.COMPONENTS_URL + 'evergreen-field/evergreen-field.html'
      };
    }
  ]);
```

This way, angular will lookup that provider by string name, then inject it in the given position in the function arguments. Using this format prevents errors during minification and should save us some time on minifying sources.

Comment (mparent61) [inline app/components/evergreen-field/evergreen-field-directive.js]: Thanks, i was missing the 'routes' param

**Comment by mparent61** [comment]
@kand @spra85 @collin

**Comment by kand** [comment]
I think this is simple enough to warrant skipping over testing.

**Comment by spra85** [comment]
We'll have you writing so much Angular before no time. :+1:

**Comment by mparent61** [comment]
:smile: :gun:

**Comment by kand** [comment]
:+1:

**Pull request merged**